src/wp-includes/comment.php

```diff
@@ -2447,9 +2448,11 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 		return false;
 	}
 
-	// Only send notifications for approved comments.
-	if ( ! isset( $comment->comment_approved ) || '1' !== $comment->comment_approved ) {
-		return false;
+	// Send notifications for approved comments and all notes.
+	if (
+		! isset( $comment->comment_approved ) ||
+		( '1' !== $comment->comment_approved && ! $is_note ) ) {
```

Comment:
In the future, it could be nice to have filters for this instead of hardcoded behavior for notes

Reply:
👍🏼  Yea, its a bit odd to have this logic here. There is a filter just above  `notify_post_author` that can be used to opt out of notifications. The status and type should probably be used to set the default that is run thru that filter instead of being checked later.